### DISCUSSION TITLE
Some more comments regarding development of GOODF

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#49] [General]
**Panel resizing: currently not possible.** 
For reasons I can only attribute to an unwanted windows update, my screen resolution was altered. The panel now runs beyond the screen and I cannot see more than about 80% of it. If I change the resoltion of my screen, my desktop gets disrupted.... again. It should be possible to make the screen adjustable to accommodate various screen resolutions simply by allowing its resizing.

**Default text in test boxes**
At it is up until now, every time text needs to be entered in any text/ numeric box, the default text has to be deleted or highlighted before any new text/ numbers can be typed. It would help if the default text automatically disappears as soon as the cursor is clicked into the box so typing of data is immediately possible.

**Image on & image off**
When images are saved on a pc, the alphabetic listing of files is 'off' first followed by 'on' after that. It may make more sense to swap the current positions of the two path boxes.

**Reorganisation of GUI elements**
For some reason, whilst creating a new odf, the GUI elements seems to have been generated randomly. I may have a slight OCD if you will and preferred if items can be re-organised in alphanumeric order or by being able to drag items up or down. It will help in identifying a particular item and may even help identify a missing element.

**Use of Arrows up/down, left/right**
Currently moving up and down the left window of the software is possible but in some places the cursor jumps to a different heading. It would be desirable to allow manouverability of the cursor up and down the list an item at a time and, with the help of L/R arrows, to move to the parent headings, closing or opening a list in the tree accordingly.

**Default font colour**
I have mentioned it before. I still think the default should be black until changed by user. Dark Red is never a default colour in any software I have used.

**Resizing of images/ elements**
jOrgan allows easy re-sizing of any element on a screen (skin). So far this is not possible with GOODF.

**Contructing and Playing VPOs**
jOrgan is particularly wonderful, versatile and very practical and user-friendly in that it is both the software to play a VPO as well as the software to construct or make changes to a VPO. One does not need any special knowledge to tweek the skin elements to suit a particular use, preference or display resolution. 

Wouldn't it be great if the same could be achieve for GO/ GOODF, simply clicking an icon to go from play to construct modes? The wheel has been invented by Sven Meiers. Now it is a matter of applying it and using it for a truck in the same way as for a car to benefit both the truck and its driver.

**Crescendo**
Last but not least, something needs to be done to allow the creation of a crescendo pedal within an odf. Currently it is an enormous challenge. I am not an idiot and I have found it daunting and challenging ..... and..... worst of all without success in spite of several tries. It is hugely frustrating!!!

**Comment by larspalo** [comment]
> Panel resizing: currently not possible.

Your issue is likely that of scaling in the OS. I am not going to implement the type of scaling GO actually uses for panels since that would make it impossible to accurately place elements with pixel perfect positions due to rounding issues. That's why I deliberately implemented the gui representation as always 100% of its actual size. If you're not happy with your display scaling - change it (in the correct place for what you really want to acheive). But I can promise you this: _it is already possible to change the size of the panels_ - but not **"scale"** them!

> Default text in test boxes

Your suggestion would make it impossible to edit existing texts, one would have to enter everything from scratch all the time when selecting a text field. **Not** going to happen! What I'm considering though, is to set focus and select the name text, just like when adding stops, whenever creating a new element. Perhaps even whenever a new object is selected, but I'm going to have to test that since it could possibly lead to accidents too (hitting any key while the textbox has focus and is selected will most likely remove anything in it - which might not always be the intention of the user).

> Image on & image off

Again this is an OS dependent issue that you can adjust by your own choice (how you want files sorted) in the dialog on your computer. I don't know if you realize it or not, but the file list is coming from the file system and sorting is up to the user, it's not forced by the software. Normally there's an arrow at each column header that allow you to sort files the way you like it. As a last resort - name the files in another way so that you get the order you want...

> Reorganisation of GUI elements

Yes, I'm planning to allow re-organisation of the user added GUI elements on the panels, just like I'm currently working on it on many of the structural organ elements. Also, sorting/re-ordering of the panels themselves and many other things. It just doesn't happen overnight though, as the code for it must be written first.

> Use of Arrows up/down, left/right

Some of the items in the tree are not possible to select (move to) in themselves - they are just there for grouping the entries in a logical manner. Thus whenever one of them is selected the focus will change (jump) to its parent instead. That's very much intended.

> Default font colour

I answer again the same as before: GOODF always uses the default colours that GrandOrgue will use for the elements. That policy is not going to change for obvious reasons that you **should** understand. I don't care what default colours are used for other software, it's what GrandOrgue uses that's important for GOODF!

> Resizing of images/ elements

GOODF uses the same approach that GrandOrgue uses, and will always do that. It's possible to set the width and height of elements - but it will not "scale" them in the manner that you likely expect, it will either "crop" or "tile" the image. GOODF is just a tool for creating organ definition files for GrandOrgue, not an image editor. Nothing that doesn't match GrandOrgue behaviour or possibilities will be done in GOODF either.

To each his own. I have personally no higher thoughts of jOrgan or how it works, but if _you_ want to use it that's up to you and I won't say anything about it. Please also respect _my_ opinions and decisions!

Regarding the crescendo, I've already expressed my opinion at https://github.com/GrandOrgue/grandorgue/issues/1624#issuecomment-1666544354 and it's likely going to be implemented in one way or another, but it most likely will not happen very quickly since there are so many other things to work on. The crescendo **is** already **possible** to do as a workaround! That you haven't figured out how, _yet_, is another thing. Just work on it and you'll eventually get it!

**Comment by Bumblebee001** [comment]
Thanks for the explanation. I did not intend to impose my ideas but merely
to highlight possible areas of improvement or development. Of course I
respect your opinions and decisions. I may not have been clear in some
instances and was not able to convey exactly what I mean. I am not good at
computer vocabulary / terminology. Neither do I know anything about
programming. So apologies for the effects of those limitations on my part.

I can go so much as a user of software with modest computer literacy and
comparing features of one versus another. I would really like to see the
best user friendly features of other software implemented in the software I
prefer to use. But then it is not up to me and the final decision is
definitely not mine either.

Thank you for taking the initiative to develop this software that overall
greatly facilitates the creation of sample sets.
